tests/core/test_iamc.py

```diff
@@ -2,214 +2,229 @@
+        run.iamc.remove(remove_data, type=_type)
+
+        # Retrieve from database
+        ret = run.iamc.tabulate(raw=raw)
+        if raw:
+            ret = ret.drop(columns=["id", "type"])
+        assert_unordered_equality(remaining_data, ret, check_like=True)
+
+        ts_after_delete = platform.backend.iamc.timeseries.tabulate(
+            join_parameters=True
+        )
+        all_dp_after_delete = platform.backend.iamc.datapoints.tabulate()
+        assert set(ts_after_delete["id"].unique()) == set(
+            all_dp_after_delete["time_series__id"].unique()
+        )
+
+        # == Partial Update / Partial Addition ==
+        # Update all data values
+        data["value"] = -9.9
+
+        # Results in a half insert / half update
+        run.iamc.add(data, type=_type)
+
+        # Retrieve from database
+        ret = run.iamc.tabulate(raw=raw)
+        if raw:
+            ret = ret.drop(columns=["id", "type"])
+        assert_unordered_equality(data, ret, check_like=True)
+
+        # == Full Removal ==
+        # Remove all data
+        remove_data = data.drop(columns=["value"])
+        run.iamc.remove(remove_data, type=_type)
+
+        # Retrieve from database
+        ret = run.iamc.tabulate(raw=raw)
+        assert ret.empty
+
+
+class TestCoreIamcBig:
+    medium = MediumIamcDataset()
+
+    @pytest.mark.parametrize(
+        "filters, run, exp_len",
+        (
+            (dict(variable={"name": "Variable 4"}), ("Model 1", "Scenario 1", 1), 50),
+            (
+                dict(
+                    variable={"name": "Variable 4"},
+                    unit={"name": "Unit 4"},
+                ),
+                ("Model 1", "Scenario 1", 1),
+                50,
+            ),
+            (
+                dict(
+                    variable={"name__like": "Variable *"},
+                    unit={"name__like": "Unit *"},
+                ),
+                ("Model 2", "Scenario 2", 1),
+                280,
+            ),
+            (
+                dict(variable={"name__in": ["Variable 4", "Variable 24"]}),
+                ("Model 1", "Scenario 1", 1),
+                50,
+            ),
+            (dict(variable="Variable 24"), ("Model 1", "Scenario 1", 1), 0),
+            (
+                dict(variable="Variable 54", unit="Unit 54"),
+                ("Model 8", "Scenario 0", 1),
+                50,
+            ),
+            (
+                dict(variable=["Variable 54", "Variable 25", "Variable 4"]),
+                ("Model 1", "Scenario 1", 1),
+                50,
+            ),
+        ),
+    )
+    def test_run_tabulate_with_filter_raw(
+        self, platform: ixmp4.Platform, filters, run, exp_len
+    ):
+        self.medium.load_dataset(platform)
+        run = platform.runs.get(*run)
+        obs = run.iamc.tabulate(raw=True, **filters)
+        assert len(obs) == exp_len
+        # assert_unordered_equality(obs, exp, check_like=True)
```

Comment:
The used much less test data before but checked the entire dataframe. I removed the assertion because the entire dataframe is already tested in the data layer and this test is more about the run-bound filtering API.

Reply:
Removed the comment.